test/sql/filter/test_filter_clause.test

```diff
@@ -0,0 +1,273 @@
+# name: test/sql/filter/test_filter_clause.test
+# description: Test aggregation with filter clause
```

Comment:
I would like some more test cases:

* Query with many different filter clauses (e.g. 5 aggregates, 5 different filters)
* Filter with some more complex aggregates: `COVAR_POP` (multiple input columns), `STRING_AGG` (strings) and `ARRAY_AGG` (lists)
* DISTINCT aggregates

Also; looking at these tests I would not be surprised if all of them use the perfect hash aggregate. You can force the regular hash aggregate to be used by using very spaced out groups (e.g. `[0, 10000000, 20000000, ....]`).